jerry-core/jerry.c

```diff
@@ -774,17 +579,20 @@ jerry_set_array_index_value (jerry_object_t *array_obj_p, /**< array object */
 bool
 jerry_get_array_index_value (jerry_object_t *array_obj_p, /**< array object */
                              jerry_length_t index, /**< index to be written */
-                             jerry_value_t *value_p) /**< output value at index */
+                             jerry_value_t *value_p) /**< [out] value at index */
 {
   ecma_string_t *str_idx_p = ecma_new_ecma_string_from_uint32 ((uint32_t) index);
   ecma_value_t get_completion = ecma_op_object_get (array_obj_p, str_idx_p);
-  JERRY_ASSERT (!ecma_is_value_error (get_completion));
-  jerry_convert_ecma_value_to_api_value (value_p, get_completion);
-
-  ecma_free_value (get_completion);
   ecma_deref_ecma_string (str_idx_p);
 
-  return true;
+  *value_p = get_completion;
+
+  if (!ecma_is_value_error (get_completion))
+  {
+    return true;
+  }
+
+  return false;
```

Comment:
Btw I am not sure people expects that the value stores the exception. Perhaps we should free the exception and return with undefined if unsuccessful.